src/common/com/intellij/plugins/haxe/model/HaxeParameterModel.java

```diff
@@ -90,8 +95,16 @@ public HaxeTypeTag getTypeTagPsi() {
     return parameter.getTypeTag();
   }
 
-  public SpecificTypeReference getType() {
-    return HaxeTypeResolver.getTypeFromTypeTag(getTypeTagPsi());
+  public ResultHolder getType() {
+    return getType(null);
+  }
+
+  public ResultHolder getType(@Nullable HaxeGenericResolver resolver) {
+    if (resolver != null) {
+      ResultHolder resolved = resolver.resolve(getType(null).getType().toStringWithoutConstant());
```

Comment:
This piece of code really would be easier to read:
```haxe
ResultHolder typeResult = getType(null);
ResultHolder resolved = resolver.resolve(typeResult.getType().toStringWithoutConstant());
```
The way my mind works, I read getType(null) to call a local function, and the following getType() I also presumed to be the local function -- which didn't make sense.  It took at least a minute just to figure out what was going on in that line; basically that the return value of getType(null) was not the same type as 'this'.